src/client.rs

```diff
@@ -223,85 +223,109 @@ mod tests {
+                        "last_modified": 100
+                    })
+                ])
+            },
+            Test {
+                test_description: "test_get_fails_if_verification_fails_with_invalid_signature_error",
+                client: Client::create_with_server_collection(&mock_server.url(""), "url-classifier-skip-urls", Some(Box::new(VerifierWithInvalidSignatureError{}))),
+                changeset_response_body: r#"{
+                    "metadata": {},
+                    "changes": [],
+                    "timestamp": 0
+                }"#,
+                expected_result: Err(ClientError::Error{name: "invalid signature error".to_owned()})
+            }
+        ];
 
-    #[tokio::test]
-    #[with_mock_server]
-    async fn test_get_fails_if_verification_fails_with_invalid_signature_error() {
-        let get_changeset_mock = mock(GET, "/buckets/main/collections/url-classifier-skip-urls/changeset")
-        .expect_query_param("_expected", "0")
-        .return_status(200)
-        .return_header("Content-Type", "application/json")
-        .return_body(
+        for test in &tests {
+
+            debug!("test name: {}", test.test_description);
+
+            let mut get_latest_change_mock = Mock::new()
+            .expect_method(GET)
+            .expect_path("/buckets/monitor/collections/changes/records")
+            .expect_query_param("bucket", "main")
+            .expect_query_param("collection", "url-classifier-skip-urls")
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_header("etag", "12345")
+            .return_body(
             r#"{
-                "metadata": {},
-                "changes": [],
-                "timestamp": 0
+                "data": [
+                    {
+                        "id": "123",
+                        "last_modified": 9173,
+                        "bucket":"main",
+                        "collection":"url-classifier-skip-urls",
+                        "host":"localhost:5000"
+                    }
+                ]
             }"#
-        ).create();
+            ).create_on(&mock_server);
+
+            let mut get_changeset_mock = Mock::new()
+            .expect_method(GET)
+            .expect_path("/buckets/main/collections/url-classifier-skip-urls/changeset")
+            .expect_query_param("_expected", &EXPECTED_VERSION.to_string())
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(test.changeset_response_body).create_on(&mock_server);
+
+            let actual_result = test.client.get().await;
+            assert_eq!(actual_result, test.expected_result);
+            assert_eq!(1, get_changeset_mock.times_called());
+            assert_eq!(1, get_latest_change_mock.times_called());
+
+            get_changeset_mock.delete();
+            get_latest_change_mock.delete();
+        }
```

Comment:
You should add a test here to assert the behaviour when bucket/collection are unknown on the server

Reply:
done